tests/unit/test_models.py

```diff
@@ -50,5 +55,78 @@ def test_invalid(self):
             TransferType.from_str(self.invalid_type)
 
 
+class TestTransfer(unittest.TestCase):
+
+    def test_add_slippage(self):
+        solver = Address.zero()
+        transfer = Transfer(
+            token_type=TokenType.NATIVE,
+            token_address=None,
+            receiver=solver,
+            amount=1.0
+        )
+        positive_slippage = SolverSlippage(
+            solver_name="Test Solver",
+            solver_address=solver,
+            amount_wei=10 ** 18 // 2
+        )
+        negative_slippage = SolverSlippage(
+            solver_name="Test Solver",
+            solver_address=solver,
+            amount_wei=-10 ** 18 // 2
+        )
+        transfer.add_slippage(positive_slippage)
+        self.assertAlmostEqual(transfer.amount, 1.5, delta=0.0000000001)
+        transfer.add_slippage(negative_slippage)
+        self.assertAlmostEqual(transfer.amount, 1.0, delta=0.0000000001)
+
+    def test_errors(self):
+        transfer = Transfer(
+            token_type=TokenType.NATIVE,
+            token_address=None,
+            receiver=ONE_ADDRESS,
+            amount=1.0
+        )
+        slippage = SolverSlippage(
+            solver_name="Test Solver",
+            solver_address=TWO_ADDRESS,
+            amount_wei=0
+        )
+        with self.assertRaises(AssertionError):
```

Comment:
Yea we can assert the exact error for sure, but it might be a built uglier. My first thoughts would be to catch the error itself and then match the error string. Definitely possible though, I can look into it if you like and post here what I find.

Reply:
Here is an example of testing raised errors by specific error:

```py
    def test_invalid(self):
        with self.assertRaises(ValueError) as err:
            TransferType.from_str(self.invalid_type)
        self.assertEqual(
            str(err.exception),
            f"No TransferType {self.invalid_type}!"
        )
```